pinot-core/src/main/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactory.java

```diff
@@ -276,7 +282,17 @@ public void setThreadResourceUsageProvider(ThreadResourceUsageProvider threadRes
     }
 
     @Override
+    @Deprecated
     public void updateQueryUsageConcurrently(String queryId, long cpuTimeNs, long memoryAllocatedBytes) {
+    }
+
+    @Override
+    public void updateQueryUsageConcurrently(String queryId, long cpuTimeNs, long memoryAllocatedBytes,
+                                             TrackingScope trackingScope) {
+      if (trackingScope != TrackingScope.QUERY) {
```

Comment:
Is this change related to this feature ?

Reply:
Yes, this is needed to make sure the existing query killing works. Once we make the ResourceUsageAccountant the default, this entire file can be nuked. Some explanation for why this is needed - https://github.com/apache/pinot/pull/15798#discussion_r2162718304

Reply:
I think I finally the grok the code. Watcher task calls two aggregators that will both decide to kill queries independently - one after the other. `QueryAggregator` continues to be triggered similar to `PerQuery...WatcherTask`. `WorkloadAggregator` is triggered by completely different rules.